build-support/bin/cherry_pick/helper.js

```diff
@@ -0,0 +1,174 @@
+    const { data: pull } = await this.octokit.rest.pulls.get({
+      ...this.#pull_context,
+    });
+
+    if (pull.milestone == null) {
+      const { data: workflows } =
+        // NB: GitHub doesn't give you the URL to the workflow, or the filename, when running a workflow.
+        await this.octokit.rest.actions.listRepoWorkflows({
+          ...this.context.repo,
+        });
+      const workflow_url = workflows.workflows.filter(
+        (workflow) => workflow.name === this.context.workflow
+      )[0].url;
+
+      await this.#add_failed_label();
+      await this.#add_comment(
+        `I was unable to cherry-pick this PR; the milestone seems to be missing.
+
+@${
+          pull.merged_by.login
+        }: Please add the milestone to the PR and re-run the [Auto Cherry-Picker job](${workflow_url}) using the "Run workflow" button.
+
+${this.#run_link}`
+      );
+      this.core.setFailed(`PR missing milestone.`);
+      return null;
+    }
+
+    const milestones = await this.#get_relevant_milestones(
+      pull.milestone.title
+    );
+    return {
+      pr_num: this.pull_number,
+      merge_commit: pull.merge_commit_sha,
+      milestones: milestones,
+    };
+  }
+
+  async cherry_pick_finished(merge_commit_sha, matrix_info) {
+    // NB: Unfortunately, we can't have the cherry-pick job use outputs to send the PR numbers, since
+    //  it uses a matrix and currently GitHub doesn't support matrix job outputs well
+    //  (See https://github.com/orgs/community/discussions/26639#discussioncomment-3252675).
+    const infos = await Promise.all(
+      matrix_info.map(async ({ branch_name, milestone }) => {
+        return this.octokit.rest.pulls.list({
+          ...this.#pull_context,
+          state: "open",
+          head: `${this.context.repo.owner}:${branch_name}`,
+        });
+      })
+    ).then((all_pulls) =>
+      all_pulls.map(({ data: pulls }, index) => {
+        return { ...matrix_info[index], pr_url: (pulls[0] || {}).html_url };
+      })
+    );
+
+    let any_failed = false;
+    let comment_body = "";
+    infos.forEach(({ pr_url, milestone, branch_name }) => {
+      if (pr_url === undefined) {
+        any_failed = true;
+        comment_body += `## :x: ${milestone}
+
+I was unable to cherry-pick this PR to ${milestone}, likely due to merge-conflicts.
+
+<details>
+<summary>Steps to Cherry-Pick locally</summary>
+
+To resolve:
+1. (Ensure your git working directory is clean)
+2. Run the following script to reproduce the merge-conflicts:
+    \`\`\`bash
+    git checkout https://github.com/pantsbuild/pants main \\
+      && git pull \\
+      && git fetch https://github.com/pantsbuild/pants ${milestone} \\
+      && git checkout -b ${branch_name} FETCH_HEAD \\
+      && git cherry-pick ${merge_commit_sha}
+    \`\`\`
+3. Fix the merge conflicts and commit the changes
+4. Run \`build-support/bin/cherry_pick/make_pr.sh -- --pull-number="${this.pull_number}" --milestone="${milestone}"\`
```

Comment:
I think the script's option parsing might've evolved (to be simpler)?
```suggestion
4. Run \`build-support/bin/cherry_pick/make_pr.sh "${this.pull_number}" "${milestone}"\`
```